main.go

```diff
@@ -12,13 +12,21 @@ import (
 
 func makeHandler(name string, fn func(http.ResponseWriter, *http.Request, string, string)) http.HandlerFunc {
```

Comment:
To clarify my concern/question, you've added GitHub-specific rules to a func named `makeHandler`. If that func happens to be used to generate, for example, non-GitHub-related handlers, then it's not correct (because those non-GitHub-related handlers may not want to accept periods at all).

I'm not familiar with this codebase so I can only go based on the name of func and its documentation, which doesn't make it clear.